pinot-core/src/test/java/com/linkedin/pinot/core/realtime/impl/dictionary/MutableDictionaryTest.java

```diff
@@ -0,0 +1,158 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.core.realtime.impl.dictionary;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import org.apache.commons.lang.RandomStringUtils;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+
+/**
+ * Unit test for mutable dictionary implementations.
+ */
+public class MutableDictionaryTest {
+
+  private static final String COLUMN_NAME = "testColumn";
+  private static final int DICTIONARY_SIZE = 1000;
+  private Random _random;
+
+  @BeforeClass
+  public void setup() {
+    _random = new Random();
+  }
+
+  @Test
+  public void testInt() {
```

Comment:
test for non-existing values,
inRange method for float
and get on empty dictionary